### ISSUE TITLE
[Docs] Integrate Move Book into Aptos.dev

### ISSUE THREAD

**Issue by davidiw** [aptos-labs/aptos-core#6375]
# Aptos Documentation Issue

https://github.com/move-language/move/tree/aptos-main/language/documentation/book

Let's get this hosted somewhere and start referring to it instead of the legacy move book.

**Comment by clay-aptos** [comment]
I have updated the internal thread on this seeking input. Thanks, all.

**Comment by clay-aptos** [comment]
Adding the points for tracking here:

1. This is indeed already the latest:
https://move-language.github.io/move/introduction.html

2. And that it is already populated from our:
https://github.com/move-language/move/blob/aptos-main/language/documentation/book/src/introduction.md

3. Which appears to indeed be actively updated?:
https://github.com/move-language/move/commits/aptos-main/language/documentation/book/src

4. From the aforementioned thread and this issue, I am determining if our goal is to get the `aptos-main` content integrated in to [Aptos.dev](http://aptos.dev/) (via the `Aptos-core` repo):
https://github.com/aptos-labs/aptos-core/issues/6375

5. If so, will we expect the continued contributions from the community?

6. Also, I have an open question regarding an internal workshop that leverages:
https://github.com/move-language/move/tree/main/language/documentation/tutorial

What is our plan for that? I proposed we excerpt from the private workshop material into a distinct tutorial that links into the original tutorial for existing concepts, leaving our content to highlight the deltas. What does the team think? Thanks!

**Comment by wrwg** [comment]
1. Yes, but this can change any time someone makes an update on the book in _any_ branch. The current bug in the book generation is that the published book will be automatically updated from any merged .md in any branch

3. Yes we are updating the docs in our branch.

4. We want to get the entire book generated from aptos-main and aptos-main only, and host it directly on our doc site. Would be that technically a problem? I'm not acquainted with github books, does the Move team need to look into this or does doc team handle this?

5. We expect people contribute to the book in `main` branch in move repo, and we will cherry pick this into `aptos-main`.

6. We should totally copy the tutorial because the tooling used in the original is not compatible with Aptos

**Comment by clay-aptos** [comment]
Thanks so much, @wrwg ! This all makes sense, and I can definitely help with the Move Book. I started looking into:
https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site

The issue is two-fold:
1. The books appear to be standalone, a la the current book. So embedding it in Aptos.dev might result in dual navigation.
2. Since Aptos.dev is built from `aptos-core`, we would need to merge the contents from two repos.

The question is, do we want an actual standalone GitHub book or the contents to appear seamlessly in Aptos.dev? I am not sure both are possible.

If the latter, it would be wonderful to have the contents in the `aptos-core` repo, but we may be able to work around it (`rsync` anyone?).

If someone can grant me admin access to the Move repo, I can try out the above. Or someone with that access can conduct the steps. Please let me know how to proceed. Thanks again!

**Comment by wrwg** [comment]
> The question is, do we want an actual standalone GitHub book or the contents to appear seamlessly in Aptos.dev? I am not sure both are possible.

We really have all freedom here to do what we think is the best, Clay. Standalone github book is not a goal per-se. If you can come up with a better way to integrate the markdown of the book into our docset, that would be great.

From source management viewpoint, there is an advantage if we keep this in the Move repo, however. This is because then it is easier to cherry pick changes to the shared Move docs  on `main` to our version, because its in the same repo at the same location. There is also the aspect of translations, which are a community effort in the Move repo, and we may want to take advantage off. If we move the sources to our repo all this might be lost. However, how we integrate the .md into our docset is all our decision.

**Comment by clay-aptos** [comment]
> > The question is, do we want an actual standalone GitHub book or the contents to appear seamlessly in Aptos.dev? I am not sure both are possible.
> 
> We really have all freedom here to do what we think is the best, Clay. Standalone github book is not a goal per-se. If you can come up with a better way to integrate the markdown of the book into our docset, that would be great.
> 
> From source management viewpoint, there is an advantage if we keep this in the Move repo, however. This is because then it is easier to cherry pick changes to the shared Move docs on `main` to our version, because its in the same repo at the same location. There is also the aspect of translations, which are a community effort in the Move repo, and we may want to take advantage off. If we move the sources to our repo all this might be lost. However, how we integrate the .md into our docset is all our decision.

Thanks, Wolfgang. This helps plenty. @davidiw, what are your preferences here: standalone book or direct integration into Aptos.dev?

**Comment by clay-aptos** [comment]
I am including input from one of our community members I met at the NFT summit on our Move docs we should attempt to address here:

"i’m really used to Rustdocs with «renderable» examples, but we’re talking Move here. the token module is just a bit too much to process in one go, and without embedded examples it"

Perhaps we can add more examples to the book itself? Team, let me know if you want to speak with this community member.

**Comment by clay-aptos** [comment]
Assuming we want integration in Aptos.dev (as the standalone book already exists and is up-to-date) and want to keep the `move-language` repository as the source of truth, I believe this tool should help:
https://github.com/google/copybara

As described here:
https://blog.kubesimplify.com/moving-code-between-git-repositories-with-copybara

And implemented as a GitHub action here:
https://github.com/olivr/copybara-action
https://blog.kubesimplify.com/moving-code-between-git-repositories-with-copybara

Anybody want to work with me on it? Regardless, I will continue digging in starting with a one-time import to the site to simply see how hard that is, what customization it requires, and whether the team thinks we should proceed with the automation copybara offers.

**Comment by clay-aptos** [comment]
@0xchloe and @wrwg , from above (my question 6, Wolfgang's answer 5) I am trying to determine our intent for the forked `move/language/documentation/tutorial` in our currently private `nft-tutorial`.

A. Did we want to include those contents directly in Aptos.dev?
B. Expose them where they are?
C. Try to excerpt the deltas in some fashion?

Thanks!

**Comment by wrwg** [comment]
It would be really great to have the generic Move tutorial adapted and actually working with Aptos. The NFT tutorial might be a story by itself, but the generic one serves its purpose, and I guess it also has a wider scope (like Move Prover etc.)

I can ask in the Move team whether someone wants to look at this.

I used copybara at Google. Sure, if you have to... But take at look at the submit history of the Move book. The changes are marginal over a timeline of +2 years. I think we can easily keep this manually in sync. 

So in a nutshell: I would just copy everything (book, tutorial) and adapt it for our needs. I think Sui does the same.

**Comment by clay-aptos** [comment]
Hi David, Wolfgang, and the Move team,

I have included the Move Book contents in Aptos.dev with:
https://github.com/aptos-labs/aptos-core/pull/6739

See it staged at:
https://deploy-preview-6739--aptos-developer-docs.netlify.app/guides/move-guides/book/SUMMARY

It was actually quite simple once I identified an issue or two with the source materials, namely a malformed link in `generics.md` that uses `abilities.html` when it should instead use `abilities.md` (Markdown extension) for our builds to work.

Note, we should also fix the “encounterd” misspelling in `standard-library.md` of the book upstream. Point me at the files and branch if you want me to do it.

I have determined we don’t need headers for docusaurus, which makes things even easier. So we merely need to:

1. Check out and copy the latest Markdown files into place.
2. Delete the `overview.md` file as not used (and filled with conflicting style issues)
3. Add any new files to navigation (sounds rare).
4. Fix new build issues (hopefully avoidable).

Regarding the NFT Tutorial, we can either expose it where it is and update the book’s link to it. Or we can attempt to ingest its contents into Aptos.dev directly as we have the book, if you like.

I will leave the central Move tutorial to the team to decide its fate. Please let me know what you think of the Move Book and our NFT Tutorial. Thanks!

**Comment by clay-aptos** [comment]
Reopening to announce the book and pursue the NFT Tutorial. :-)

**Comment by clay-aptos** [comment]
Here is the announcement PR:
https://github.com/aptos-labs/aptos-core/pull/6758

**Comment by clay-aptos** [comment]
Reopening to address the NFT Tutorial discussed in:
https://github.com/aptos-labs/aptos-core/issues/6375#issuecomment-1441208359

**Comment by clay-aptos** [comment]
@wrwg and team, here are the fixes to the book files:
https://github.com/move-language/move/pull/941
https://github.com/move-language/move/pull/942

**Comment by clay-aptos** [comment]
Hi @wrwg , you can see a first attempt at migrating the forked Move tutorial in:
https://github.com/aptos-labs/aptos-core/pull/6851

Sadly, the links to subdirectories don't work (but links to files do) as Aptos.dev isn't GitHub. :-)

So we should figure out where that code should live and link to it with fully qualified links (ex. https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/)

Please let me know what you think!

**Comment by clay-aptos** [comment]
Descoping and closing this out as @wrwg has another issue open for the Move tutorial rework. Wolfgang, please feel free to add me to that list. Thanks, all.
